smack-extensions/src/main/java/org/jivesoftware/smackx/geoloc/GeoLocationManager.java

```diff
@@ -89,52 +89,60 @@ public static boolean isGeoLocationMessage(Message message) {
     }
 
     /**
-     * Returns the geolocation node in PubSub.
+     * Send geolocation through the PubSub node.
      * 
-     * @return the geolocation PubSub node
+     * @param geoLocation
      * @throws InterruptedException
      * @throws NotConnectedException
      * @throws XMPPErrorException
      * @throws NoResponseException
      */
-    public LeafNode getNode()
+    public void sendGeolocation(GeoLocation geoLocation)
             throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
-        PubSubManager pubSubManager = PubSubManager.getInstance(connection());
-
-        try {
-            pubSubManager.createNode(GeoLocation.NAMESPACE);
-        } catch (Exception e) {
-        }
-
-        return pubSubManager.getNode(GeoLocation.NAMESPACE);
+        getNode().send(new PayloadItem<GeoLocation>(geoLocation));
     }
 
     /**
-     * Send geolocation through the PubSub node.
+     * Send empty geolocation through the PubSub node.
      * 
-     * @param geoLocation
      * @throws InterruptedException
      * @throws NotConnectedException
      * @throws XMPPErrorException
      * @throws NoResponseException
      */
-    public void sendGeolocation(GeoLocation geoLocation)
+    public void stopPublishingGeolocation()
             throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
-        getNode().send(new PayloadItem<GeoLocation>(geoLocation));
+        GeoLocation emptyGeolocation = new GeoLocation.Builder().build();
+        getNode().send(new PayloadItem<GeoLocation>(emptyGeolocation));
     }
 
     /**
-     * Send empty geolocation through the PubSub node.
+     * Returns the geolocation node in PubSub.
      * 
+     * @return the geolocation PubSub node
      * @throws InterruptedException
      * @throws NotConnectedException
      * @throws XMPPErrorException
      * @throws NoResponseException
      */
-    public void stopSendingGeolocation()
+    public LeafNode getNode()
             throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
-        GeoLocation emptyGeolocation = new GeoLocation.Builder().build();
-        getNode().send(new PayloadItem<GeoLocation>(emptyGeolocation));
+        PubSubManager pubSubManager = PubSubManager.getInstance(connection());
+        LeafNode node;
+
+        try {
+            node = pubSubManager.getNode(GeoLocation.NAMESPACE);
+        } catch (XMPPErrorException notExistsException) {
+            // the node does not exist
```

Comment:
I think an XMPPErrorException could be thrown in other cases too, so you have to check the exact error condition. You most likely want to check for item-not-found (XEP-0060 § 6.1.3.12) to check if you need to create the node, and otherwise rethrow the exception.